File: maximum-depth-of-binary-tree/maximum-depth-of-binary-tree.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */

int maxDepth(struct TreeNode* root){
    int r = 0;

    if (root == NULL) {
        return r;
    }

    int ld = maxDepth(root->left);
    int rd = maxDepth(root->right);
    if (ld > rd) {
        r = ld + 1;
    } else {
        r = rd + 1;
    }
    return r;
}

File: maximum-depth-of-binary-tree/maximum-depth-of-binary-tree.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func maxDepth(root *TreeNode) int {
    if root == nil {
        return 0;
    }

    ld := maxDepth(root.Left)
    rd := maxDepth(root.Right)

    if ld > rd {
        return ld + 1
    } else {
        return rd + 1
    }
}
